lib/Corres_UL.thy

```diff
@@ -246,10 +246,10 @@ lemma corres_split':
 text \<open>Derivative splitting rules\<close>
 
 lemma corres_split:
-  assumes y: "\<And>rv rv'. r' rv rv' \<Longrightarrow> corres_underlying sr nf nf' r (R rv) (R' rv') (b rv) (d rv')"
   assumes x: "corres_underlying sr nf nf' r' P P' a c"
+  assumes y: "\<And>rv rv'. r' rv rv' \<Longrightarrow> corres_underlying sr nf nf' r (R rv) (R' rv') (b rv) (d rv')"
   assumes    "\<lbrace>Q\<rbrace> a \<lbrace>R\<rbrace>" "\<lbrace>Q'\<rbrace> c \<lbrace>R'\<rbrace>"
-  shows      "corres_underlying sr nf nf' r (P and Q) (P' and Q') (a >>= (\<lambda>rv. b rv)) (c >>= (\<lambda>rv'. d rv'))"
+  shows      "corres_underlying sr nf nf' r (P and Q) (P' and Q') (a >>= b) (c >>= d)"
```

Comment:
These `%rv. b rv` patterns are often intentional. Even though the internal term representation primarily uses de Bruijn indices, it remembers bound variable names to use as hints to use in printing. These often end up being visible in the goal. (Every variable name that shows as green is one of these bound variables, and its name is derived from this hint.)
When the higher-order unifier has to generate a lambda abstraction, I guess it sometimes generates such a hint from one of the patterns it's matching. I don't know enough details to know how that works, or whether that would happen here, or under what circumstances that `rv` might end up being visible in a goal. But I think the intention is that if it does end up being visible, it shows as `rv` rather than `x7`. I think @Xaphiosis is quite keen on this, so perhaps he knows more.

Reply:
I missed this. Yes, this should definitely be left in. I thought it's also so that if you have:

```
do
  tptr <- f x;
  tcb <- g tptr;
  h
od
```

and you split these with split rules, then you get `tptr` and `tcb` as your variables, instead of what's  used *inside* the split rule itself or some other rando as @mbrcknl mentioned, or usually `rv`, `rva`, 'rvaa` etc. It's lame to do a split, immediately followed by rename_tac when the names we want are already in the program. You'll see a bunch of this in fastpath proofs. @lsf37 will have the Isabelle internals angle here, I only have the pragmatic attitude. Still vote to keep this schema alive.

Reply:
Yes, also has my vote. I have forgotten the internals, but exactly that was the intention behind the eta expansion here.

Reply:
Oh, I didn't know that this sort of expansion helped with names like that. I definitely agree that we want to keep it in then, especially since I don't even remember removing this in the first place.